common/flatpak-dir.c

```diff
@@ -2127,10 +2131,23 @@ flatpak_dir_pull_extra_data (FlatpakDir          *self,
 
       /* TODO: Download to disk to support resumed downloads on error */
 
-      ensure_soup_session (self);
-      bytes = flatpak_load_http_uri (self->soup_session, extra_data_uri, NULL, NULL,
-                                     extra_data_progress_report, &extra_data_progress,
-                                     cancellable, error);
+      base_dir = flatpak_get_user_base_dir_location ();
+      extra_local_dir = g_file_get_child (base_dir, "extra-data");
+      extra_local_file = g_file_get_child (g_file_get_child (extra_local_dir, extra_data_sha256), extra_data_name);
```

Comment:
This leaks the intermediate child.
Use `flatpak_build_file (base_dir, "extra-data", extra_data_sha256, extra_data_name, NULL)` instead.